tests/js/client/load-balancing/load-balancing-key-generator-noauth-cluster.js

```diff
@@ -140,15 +182,40 @@ function KeyGeneratorSuite () {
           let result = sendRequest('POST', url, /*payload*/ {}, {}, i % 2 === 0);
           assertEqual(result.status, 202);
           let key = result.body._key;
-          assertTrue(key > lastKey || lastKey === null, { key, lastKey });
+          assertTrue(key > lastKey || lastKey === null, {key, lastKey});
           lastKey = key;
         }
       } finally {
         db._useDatabase("_system");
         db._dropDatabase(cn);
       }
     },
-    
+
+    testAutoincrementOnMultipleShards: function () {
+      // check that the generated keys are sequential when we send the requests
+      // via multiple coordinators.
+      try {
+        db._create(cn, {numberOfShards: 5, keyOptions: {type: "autoincrement"}});
+      } catch (error) {
+        assertTrue(error.message === "the specified key generator is not supported for collections with more than one shard");
+      } finally {
+        db._drop(cn);
+      }
+    },
+
+    testAutoincrementOnOneShard: function () {
+      db._createDatabase(cn, {sharding: "single"});
```

Comment:
OneShard is only available in the cluster && enterprise edition, so the test needs to be restricted to that combination.